extensions/notebook/src/book/bookModel.ts

```diff
@@ -0,0 +1,221 @@
+						type: BookTreeItemType.ExternalLink,
+						treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+						untitled: this.openAsUntitled
+					},
+						{
+							light: this._extensionContext.asAbsolutePath('resources/light/link.svg'),
+							dark: this._extensionContext.asAbsolutePath('resources/dark/link_inverse.svg')
+						}
+					);
+
+					notebooks.push(externalLink);
+				} else {
+					let pathToNotebook = path.join(root, 'content', sections[i].url.concat('.ipynb'));
+					let pathToMarkdown = path.join(root, 'content', sections[i].url.concat('.md'));
+					// Note: Currently, if there is an ipynb and a md file with the same name, Jupyter Books only shows the notebook.
+					// Following Jupyter Books behavior for now
+					if (fs.existsSync(pathToNotebook)) {
+						let notebook = new BookTreeItem({
+							title: sections[i].title,
+							root: root,
+							tableOfContents: tableOfContents,
+							page: sections[i],
+							type: BookTreeItemType.Notebook,
+							treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+							untitled: this.openAsUntitled
+						},
+							{
+								light: this._extensionContext.asAbsolutePath('resources/light/notebook.svg'),
+								dark: this._extensionContext.asAbsolutePath('resources/dark/notebook_inverse.svg')
+							}
+						);
+						notebooks.push(notebook);
+						if (this.openAsUntitled) {
+							this._allNotebooks.set(path.basename(pathToNotebook), notebook);
+						}
+						else {
+							this._allNotebooks.set(pathToNotebook, notebook);
+						}
+					} else if (fs.existsSync(pathToMarkdown)) {
+						let markdown = new BookTreeItem({
+							title: sections[i].title,
+							root: root,
+							tableOfContents: tableOfContents,
+							page: sections[i],
+							type: BookTreeItemType.Markdown,
+							treeItemCollapsibleState: vscode.TreeItemCollapsibleState.Collapsed,
+							untitled: this.openAsUntitled
+						},
+							{
+								light: this._extensionContext.asAbsolutePath('resources/light/markdown.svg'),
+								dark: this._extensionContext.asAbsolutePath('resources/dark/markdown_inverse.svg')
+							}
+						);
+						notebooks.push(markdown);
+					} else {
+						let error = localize('missingFileError', 'Missing file : {0}', sections[i].title);
+						this._errorMessage = error;
+						vscode.window.showErrorMessage(error);
+					}
+				}
+			} else {
+				// TODO: search functionality (#6160)
+			}
+		}
+		return notebooks;
+	}
+
+	/**
+	 * Recursively parses out a section of a Jupyter Book.
+	 * @param array The input data to parse
+	 */
+	private parseJupyterSection(array: any[]): IJupyterBookSection[] {
+		try {
+			return array.reduce((acc, val) => Array.isArray(val.sections) ? acc.concat(val).concat(this.parseJupyterSection(val.sections)) : acc.concat(val), []);
+		} catch (error) {
+			throw localize('Invalid toc.yml', 'Error: {0} has an incorrect toc.yml file', array[0].title); //need to find a way to get title.
```

Comment:
not sure that i love that we're just throwing here. can we show an error message instead?